docs/aws/audit/elbmonitoring/rules/elb_idle_remediation.mdx

```diff
@@ -1,6 +1,235 @@
+   - Use the **Resource Groups** feature to filter and manage ELBs based on tags, making it easier to identify and review potentially idle resources.
+
+By following these steps, you can proactively monitor and manage your ELBs to prevent idle instances from persisting in your AWS environment.
+</Accordion>
+
+<Accordion title='Using CLI'>
+To prevent having idle Elastic Load Balancers (ELBs) in AWS using the AWS CLI, you can follow these steps:
+
+1. **List All Load Balancers:**
+   First, you need to list all the load balancers in your account to identify which ones are currently active and which ones might be idle.
+   ```sh
+   aws elb describe-load-balancers --query 'LoadBalancerDescriptions[*].LoadBalancerName'
+   ```
+
+2. **Check Load Balancer Metrics:**
+   For each load balancer, check the CloudWatch metrics to determine if they are receiving any traffic. You can use the `get-metric-statistics` command to fetch metrics like `RequestCount`.
+   ```sh
+   aws cloudwatch get-metric-statistics --namespace AWS/ELB --metric-name RequestCount --dimensions Name=LoadBalancerName,Value=<LoadBalancerName> --start-time <StartTime> --end-time <EndTime> --period 3600 --statistics Sum
+   ```
+
+3. **Tagging Load Balancers:**
+   Tag your load balancers with metadata that can help you identify their purpose and usage. This can help in future audits and automated scripts to identify idle resources.
+   ```sh
+   aws elb add-tags --load-balancer-names <LoadBalancerName> --tags Key=Environment,Value=Production Key=Owner,Value=TeamA
+   ```
+
+4. **Automate Monitoring:**
+   Set up a CloudWatch alarm to monitor the `RequestCount` metric for each load balancer. If the `RequestCount` is zero for a prolonged period, trigger an alert or an automated action.
+   ```sh
+   aws cloudwatch put-metric-alarm --alarm-name "IdleELBAlarm-<LoadBalancerName>" --metric-name RequestCount --namespace AWS/ELB --statistic Sum --period 3600 --threshold 1 --comparison-operator LessThanThreshold --dimensions Name=LoadBalancerName,Value=<LoadBalancerName> --evaluation-periods 24 --alarm-actions <SNS_Topic_ARN>
+   ```
+
+By following these steps, you can proactively monitor and manage your ELBs to ensure that no idle load balancers are present in your AWS environment.
+</Accordion>
+
+<Accordion title='Using Python'>
+To prevent idle Elastic Load Balancers (ELBs) in AWS using Python scripts, you can use the Boto3 library, which is the AWS SDK for Python. Here are the steps to achieve this:
+
+### Step 1: Install Boto3
+First, ensure you have Boto3 installed. You can install it using pip if you haven't already:
+```bash
+pip install boto3
+```
+
+### Step 2: Initialize Boto3 Client
+Initialize the Boto3 client for ELB:
+```python
+import boto3
+
+# Initialize a session using Amazon EC2
+session = boto3.Session(
+    aws_access_key_id='YOUR_AWS_ACCESS_KEY',
+    aws_secret_access_key='YOUR_AWS_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>